src/modules/CurrencyModule.cpp

```diff
@@ -914,21 +915,113 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
+				victimCurrency.setCurrency(victimCurrency.currency() + amountToSteal * config.stealVictimBonusPercent / 100);
+				authorCurrency.jailTimer->start(config.stealFailedJailTime * 60 * 60 * 1000);
+
+				QString fineAmount = QString::number(amountToSteal * config.stealFinePercent / 100.0);
+				QString victimBonus = QString::number(amountToSteal * config.stealVictimBonusPercent / 100.0);
+				QString jailTime = QString::number(config.stealFailedJailTime);
+				QString output = QString(
+					":rotating_light: **You got caught!** :rotating_light:\n"
+					"*%1* has been fined **`%2 %3`** and placed in jail for %4 hours.\n"
+					"*%5* has been granted **`%6 %3`** in insurance."
+				).arg(thiefName, fineAmount, config.currencySymbol, jailTime, victimName, victimBonus);
+
+				client.createMessage(message.channelId(), output);
+			}
+			return;
+		}
+
+		else if (eventConfig.eventLowRiskLowRewardRunning)
+		{
+			double successChance = (config.lowRiskRewardStealSuccessChance / 100.0) * qExp(-0.0001 * qPow(amountToSteal, 1.5));
+
+			std::random_device device;
+			std::mt19937 prng{ device() };
+			std::discrete_distribution<> distribution{ { 1 - successChance, successChance } };
+			int roll = distribution(prng);
+
+			if (roll)
+			{
+				double penalty = amountToSteal * lowRiskRewardPenalty * 0.01;
+				victimCurrency.setCurrency(victimCurrency.currency() - (amountToSteal - penalty));
+				authorCurrency.setCurrency(authorCurrency.currency() + (amountToSteal - penalty));
+				QString num = QString::number(penalty);
+
+				QString stealAmount = QString::number(amountToSteal);
+				QString output = QString(
+					":man_detective: **Steal success!** :man_detective:\n"
+					"*%1* has discreetly stolen **`%2 %3`** from under *%4's* nose.\n"
+					"LowRiskLowReward event **PENALTY**: `%5 %3`\n"
+				).arg(thiefName, stealAmount, config.currencySymbol, victimName, num);
+				client.createMessage(message.channelId(), output);
+				return;
+			}
+			else
+			{
+				authorCurrency.setCurrency(authorCurrency.currency() - amountToSteal * config.stealFinePercent / 100);
+				victimCurrency.setCurrency(victimCurrency.currency() + amountToSteal * config.stealVictimBonusPercent / 100);
+				authorCurrency.jailTimer->start(config.stealFailedJailTime * 60 * 60 * 1000);
+
+				QString fineAmount = QString::number(amountToSteal * config.stealFinePercent / 100.0);
+				QString victimBonus = QString::number(amountToSteal * config.stealVictimBonusPercent / 100.0);
+				QString jailTime = QString::number(config.stealFailedJailTime);
+				QString output = QString(
+					":rotating_light: **You got caught!** :rotating_light:\n"
+					"*%1* has been fined **`%2 %3`** and placed in jail for %4 hours.\n"
+					"*%5* has been granted **`%6 %3`** in insurance."
+				).arg(thiefName, fineAmount, config.currencySymbol, jailTime, victimName, victimBonus);
+
+				client.createMessage(message.channelId(), output);
+			}
+			return;
+		}
```

Comment:
Code duplication here. Most of the code for the three modes - "HRHR", "LRLR" and "Normal" is similar and could be combined. Create local variables instead, set them accordingly with the conditionals, and use them for the whole *success chance and friends* stuff.

Reply:
Okay done